### PR TITLE
javascript1-week4/andrii

### PR DESCRIPTION
**Pull request by ayaroshenkocom**

### CODE REVIEW

File: 3_javascript/javascript1/week4/script.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    if (command.startsWith("Hello my name is ")) {
+        const newName = command.replace("Hello my name is ", ""); // if someone writes this twice
+        if (userName) {
+            return `We already know each other. Hi again ${userName}.`;
+        } else {
+            userName = newName;
+            return `Nice to meet you ${userName}`;
+        }
+    } else if (command === "What is my name?") {
+        return userName ? `Your name is ${userName}` : "Your name is not set.";
```
Comment (Elisha2605) [inline 3_javascript/javascript1/week4/script.js:152]: Nice use of ternary operator

File: 3_javascript/javascript1/week4/script.js [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+            return "Invalid timer duration.";
+        }
+        timer = setTimeout(() => {
+            console.log("Timer done.");
+        }, minutes * 60 * 1000); // Convert minutes to milliseconds
+        return `Timer set for ${minutes} minutes.`;
+    } else if (command === "Tell me a joke") { //Here is my command
+        const randomJoke = jokes[Math.floor(Math.random() * jokes.length)];
+        return randomJoke;
+    }
```
Comment (Elisha2605) [inline 3_javascript/javascript1/week4/script.js:198]: Great job implementing a new feature!

**Review by Elisha2605 — COMMENTED** [review]
Hey Andrii,

Great job! Every command seems to work as required. But I noticed that you might have forgotten to implement the calculation command. It says in the exercise:

Should be able to do simple math. fx `what is 3 + 3` should respond with 6. Or what is 4 * 12 should respond with 48.

Otherwise, very good job!

**Comment by ayaroshenkocom** [comment]
> Hey Andrii,
> 
> Great job! Every command seems to work as required. But I noticed that you might have forgotten to implement the calculation command. It says in the exercise:
> 
> Should be able to do simple math. fx `what is 3 + 3` should respond with 6. Or what is 4 * 12 should respond with 48.
> 
> Otherwise, very good job and also it's nice that you've added an extra command to the function.

But my code can make calculation :) 
console.log(getReply("What is 3 + 3")); // "6"
console.log(getReply("What is 4 * 12")); // "48"

Answer 6 and 48

Or do you think I implemented this incorrectly because it is not safety?

**Comment by shpomp** [comment]
> > Hey Andrii,
> > Great job! Every command seems to work as required. But I noticed that you might have forgotten to implement the calculation command. It says in the exercise:
> > Should be able to do simple math. fx `what is 3 + 3` should respond with 6. Or what is 4 * 12 should respond with 48.
> > Otherwise, very good job and also it's nice that you've added an extra command to the function.
> 
> But my code can make calculation :) console.log(getReply("What is 3 + 3")); // "6" console.log(getReply("What is 4 * 12")); // "48"
> 
> Answer 6 and 48
> 
> Or do you think I implemented this incorrectly because it is not safety?

I will chime in here. 
It looks like the calculation functionality is not implemented, as Elisha states. 
You could have double checked before replying to the comment :) 
image

**Comment by ayaroshenkocom** [comment]
> console.log(getReply("What is 3 + 3"))

It is very interesting because I have checked and did it again a minutes ago. Here is it https://skr.sh/sM6kg0UPf1J

**Comment by ayaroshenkocom** [comment]
I will check it tomorrow. Maybe it is wrong code on github. Or maybe I am just being obtuse. :) It is also can can be main reason :)

**Comment by ayaroshenkocom** [comment]
I have found the problem :) https://skr.sh/sM6Ht9E60yT I still have problems with lowercase letters. :)

**Comment by shpomp** [comment]
> I found the reason :) https://skr.sh/sM6Ht9E60yT I still have problems with lowercase letters. :)

Happens to the best of us, @ayaroshenkocom , no worries. 😅 
A good chance to improve the code and make it more robust, consider more use cases and edge cases. 
But overall you are already doing really well 👌🏽 👏🏽

**Comment by ayaroshenkocom** [comment]
Thanks everyone for your feedback and time. I have fixed the problem.

**Pull request closed without merging**